src/foam/core/console/AbstractDAOAgent.js

```diff
@@ -18,22 +18,66 @@ foam.CLASS({
     {
       name: 'of',
       factory: function() { return this.referenceDAO.of; }
-    }
+    },
+    'props',
```

Comment:
This seems a little too coupled to me. Could the DAOPrompt2 export columns and then the ones that need it could import it and use if available. You could have an AbstractColumnAwareDAOAgent that extends AbstractDAOAgent that those who need it could extend.

Reply:
good idea